server/src/game/components/cult.rs

```diff
@@ -0,0 +1,110 @@
+use crate::game::{chat::{ChatGroup, ChatMessageVariant}, phase::PhaseType, player::PlayerReference, role::{apostle::Apostle, disciple::Disciple, zealot::Zealot, RoleState}, role_list::Faction, Game};
+
+
+impl Game {
+    pub fn cult(&self)->&Cult{
+        &self.cult
+    }
+    pub fn set_cult(&mut self, cult: Cult){
+        self.cult = cult;
+    }
+}
+#[derive(Default, Clone)]
+pub struct Cult {
+    pub ordered_cultists: Vec<PlayerReference>,
+    pub sacrifices_needed: Option<u8>
+}
+impl Cult{
+    pub fn on_phase_start(self, game: &mut Game, phase: PhaseType){
+        Cult::set_ordered_cultists(self.clone(), game);
```

Comment:
Should rather be: `self.clone().set_ordered_cultists(game);`
But also, why doesn't `set_ordered_cultists` take a reference and clone it itself? Why should the caller do the cloning?

Reply:
Nvm, I get it